### PR TITLE
Allow collapsing navigator conditionals

### LINKED ISSUE #3486: Allow collapsing conditionals in the navigator

### PR DESCRIPTION
**Pull request by ruggi**
Fixes #3486 

**Problem:**

Collapsing conditional items in the navigator does not hide the branches of the conditional expression.

**Fix:**

When generating the navigator items, make conditional clauses honor the `collapsedAncestor` flag correctly.

Kapture 2023-03-28 at 13 07 40

### CODE REVIEW

**Comment by ruggi** [comment]
Note: holding off with the tests until the flakiness is resolved

**Comment by ruggi** [comment]
> Note: holding off with the tests until the flakiness is resolved

added test in https://github.com/concrete-utopia/utopia/pull/3485/commits/b80d95b9d2b70c365082945ca29e17990994a3f9

File: editor/src/components/navigator/navigator-item/navigator-item.tsx [hunk 1]
```diff
@@ -545,6 +545,7 @@ export const NavigatorItem: React.FunctionComponent<
             selected={selected && !isInsideComponent}
             onMouseDown={collapse}
             style={{ transform: 'scale(0.6)', opacity: 'var(--paneHoverOpacity)' }}
+            testId={`navigator-item-collapse-${EP.toUid(props.navigatorEntry.elementPath)}`}
```
Comment (seanparsons) [inline editor/src/components/navigator/navigator-item/navigator-item.tsx:548]: I believe these will duplicate, there is a `navigatorEntryToKey` function IIRC which as long as we don't duplicate entries should generate IDs which are unique across the navigator.

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/navigator/navigator-item/navigator-item.tsx (hunk 1 above)
Comment (ruggi) [reply editor/src/components/navigator/navigator-item/navigator-item.tsx]: like this? https://github.com/concrete-utopia/utopia/pull/3485/commits/9385c64864250120e46655db7d49e8e9d4833e18

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/navigator/navigator-item/navigator-item.tsx (hunk 1 above)
Comment (seanparsons) [reply editor/src/components/navigator/navigator-item/navigator-item.tsx]: Lovely!

**Pull request merged**